src/main/java/net/neoforged/moddevgradle/internal/jarjar/JarJarArtifacts.java

```diff
@@ -221,16 +223,16 @@ private static void collectFromComponent(ResolvedComponentResult rootComponent,
         return moduleOrCapabilityVersion(variant);
     }
 
-    private static boolean isValidVersionRange(final @Nullable String range) {
-        if (range == null) {
-            return false;
-        }
+    private static String validateVersionRange(String range, ModuleComponentSelector module) {
         try {
-            final VersionRange data = VersionRange.createFromVersionSpec(range);
-            return data.hasRestrictions() && data.getRecommendedVersion() == null && !range.contains("+");
-        } catch (InvalidVersionSpecificationException e) {
-            return false;
+            var data = VersionRange.createFromVersionSpec(range);
+            if (data.hasRestrictions() && data.getRecommendedVersion() == null && !range.contains("+")) {
+                return range;
+            }
+        } catch (InvalidVersionSpecificationException ignored) {
         }
+
+        throw new GradleException("Unsupported version constraint '" + range + "' on Jar-in-Jar dependency " + module.getModuleIdentifier());
```

Comment:
Would there be any value in forwarding the `InvalidVersionSpecificationException`? Maybe it contains a useful error message